id_manager/include/essentials/Identifier.h

```diff
@@ -13,6 +13,7 @@ class Identifier
     friend struct std::hash<essentials::Identifier>;
 
 public:
+    Identifier();
```

Comment:
What do you mean by non-sense?  The id vector would be empty and can be safely compared to other id's.

Reply:
But the semantics of an default initialised Identifier is not clear:

You introduced it for giving the AliceEngine/TeamManager the message that "no agent id was specified" when the base was initialised. So the semantics now is "Default-Initialised ID == No ID", because the internal vector is empty.

Two questions now come to my mind:
1. Does this also mean that "No ID == Invalid/Garbage ID"?
2. If we want to change the internal representation from "vector<uint8_t> _id" to "uint64_t _id", the default-initialised ID would probably become "uint64_t _id = 0;". So creating a Default-Constructor implies that we reserve a specific value for representing "No ID". I am personally against such things because C++17 introduced std::optional for such semantics.

So I actually would prefer to change the AlicaContext to take an std::optional<Identifier> as parameter and have it defaulted to nullopt, instead of introducing a default constructor and reserving certain values for special handling.

Reply:
I see what you are suggesting. Using it in the main API would mean that we expect all alica applications to use 17 onwards versions. We shouldn't be breaking that compatibility yet.

Reply:
So create another constructor for the AlicaContext without the ID parameter. Once we switch to C++17, we can remove it without breaking anything. However, the default constructor of the Identifier cannot be reverted, once it is used everywhere.

Reply:
Too messy Context+Engine will need overloaded constructors. We can skip alica engine though by using constPtr there. All this is just because we never want to allow anyone to default construct Identifier? I don't see what do we gain by not allowing it. If we switch to uint64, we might not need the Identifier class anymore. And even if we keep it, std::optional can be used internally to keep the id representation (like std::optional<uint64_t>) or we can even keep it as unique_ptr<uint64_t>.

Question:: 
- if the Identifier can't be empty, why do we have the overloaded bool operator in there? 
- What do you want to do if someone passes an empty byte vector in the constructor?
My assumption is that applications will have cases where it could possibly be empty (ex: failed deserialization of incoming messages) and without default constructor, we will be forcing them to always use either optional or reserve an id at app level.

Reply:
To document the conclusions of our meeting:
* semantics of default constructed identifiers is that of an empty/malformed id
* The operator== and operator!= needs to be adapted in order to always return false, if one of the operands is empty/default constructed
* bool operator is fine as long it has the explicit keyword